src/internal/kinesis.ts

```diff
@@ -0,0 +1,161 @@
+/**
+ * It creates a new KinesisClient object and returns it.
+ * @param {T} obj - T - The object that we're checking for the key.
+ * @param k - The number of virtual users to simulate.
+ */
+import { JSONObject } from 'k6'
+import http, { RefinedResponse, ResponseType } from 'k6/http'
+
+import { AWSClient } from './client'
+
+import { AWSConfig } from './config'
+import { AWSError } from './error'
+import { HTTPHeaders } from './http'
+import { InvalidSignatureError, SignatureV4 } from './signature'
+
+const hasKey = <T extends object>(obj: T, k: keyof any): k is keyof T =>
+    k in obj;
+
+
+
+
+/* This class is a wrapper around the AWS Kinesis client. It provides a method for each Kinesis
+operation, and each method returns a promise. */
+export class KinesisClient extends AWSClient {
+
+    signature: SignatureV4
+
+    commonHeaders: HTTPHeaders
+
+    serviceVersion: string
+
+    /**
+     * A constructor function that creates a new instance of the Kinesis class.
+     * @param {AWSConfig} awsConfig - AWSConfig - This is the configuration object that is passed to
+     * the constructor.
+     * @returns A Proxy object.
+     */
+    constructor(awsConfig: AWSConfig) {
+        super(awsConfig, 'kinesis')
+
+        this.serviceVersion = 'Kinesis_20131202'
+
+
+        this.signature = new SignatureV4({
+            service: this.serviceName,
+            region: this.awsConfig.region,
+            credentials: {
+                accessKeyId: this.awsConfig.accessKeyId,
+                secretAccessKey: this.awsConfig.secretAccessKey,
+                sessionToken: this.awsConfig.sessionToken,
+            },
+            uriEscapePath: false,
+            applyChecksum: true,
+        })
+
+
+        this.commonHeaders = {
+            'Content-Type': 'application/x-amz-json-1.1',
+        }
+
+        return new Proxy(this, {
```

Comment:
Thanks @oleiade , I have started with individual functions, then realise this would mean to either replicate entire SDK functions or have limited functionality. so, utilising documentation of AWS made sense to me.

on casing standard, I can play gimmick to make it similar, I just kept it as standard as I can see in the page.
https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_Operations.html
I can update based based on review opinion and repo standard.

Reply:
Hey @autometa101 

I appreciate your understanding. We've been working with the AWS APIs long enough now to be able to judge that there is a low consistency over there. It's understandable, considering the sheer amount of APIs they have. As a result, we've set our own rules and guidelines, mostly aligned with the rest of the k6 ecosystem.

To be able to merge the Pull Request in `main`, we will indeed require the following:
- That every service method is exposed on the client as a dedicated function (which can call a single shared function under the hood), ideally with some level of documentation (in general, we copy and adapt the one from each action type in the AWS documentation). As I mentioned before, we do that for clarity and user-friendliness but also because it makes our jobs as maintainers easier when it comes to documenting the library on k6's documentation website, for instance, 👍🏻 
- That every method name starts with a lowercase letter 🛩️ 

I imagine this could feel frustrating, but as the library grows, this is how we make sure that contributing to it and using it stays accessible 🙇🏻

Reply:
completed, created method for each essential function.